procdiscovery/pkg/inspectors/langdetect.go

```diff
@@ -22,11 +22,15 @@ func (e ErrLanguageDetectionConflict) Error() string {
 	return fmt.Sprintf("language detection conflict between %v and %v", e.languages[0], e.languages[1])
 }
 
-type inspector interface {
-	Inspect(process *process.Details) (common.ProgramLanguageDetails, bool)
+type LanguageInspector interface {
+	Inspect(process *process.Details) (common.ProgrammingLanguage, bool)
 }
 
-var inspectorsList = []inspector{
+type VersionInspector interface {
+	GetRuntimeVersion(process *process.Details, containerURL string) *version.Version
```

Comment:
nit: By returning `version.Version` here, we need to transform the version string in every inspector instead of once where it is invoked.

Did you consider simplifying the api and returning string here?